src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java

```diff
@@ -386,45 +396,33 @@ public PrivilegesEvaluatorResponse evaluate(final User user, String action0, fin
             }
         }
 
-        if (dnfofEnabled
-                && (action0.startsWith("indices:data/read/")
-                || action0.startsWith("indices:admin/mappings/fields/get")
-                || action0.equals("indices:admin/shards/search_shards"))) {
+        if (dnfofEnabled && dnfofPatterns.matcher(action0).matches()) {
 
             if(requestedResolved.getAllIndices().isEmpty()) {
                 presponse.missingPrivileges.clear();
                 presponse.allowed = true;
                 return presponse;
             }
 
-
             Set<String> reduced = securityRoles.reduce(requestedResolved, user, allIndexPermsRequiredA, resolver, clusterService);
 
             if(reduced.isEmpty()) {
-                if(dcm.isDnfofForEmptyResultsEnabled()) {
-                    if(request instanceof SearchRequest) {
-                        ((SearchRequest) request).indices(new String[0]);
-                        ((SearchRequest) request).indicesOptions(IndicesOptions.fromOptions(true, true, false, false));
-                        presponse.missingPrivileges.clear();
-                        presponse.allowed = true;
-                        return presponse;
-                    }
+                if(dcm.isDnfofForEmptyResultsEnabled() && request instanceof IndicesRequest.Replaceable) {
```

Comment:
under what condition `request instanceof IndicesRequest.Replaceable` will be `false`?

Reply:
`request`, passed into function as `final ActionRequest` does not guarantee implementing `IndicesRequest.Replaceable`, keep the check for safety reason

Reply:
It is not any `request` passed to `evaluate()`, it is request where `dnfofPatterns.matcher(action0).matches()` is `true`. I'd like to understand what other type of requests that are not `Replaceable` match that condition.

Reply:
Make sense, there maybe not.

Removing the check

Reply:
Have you confirmed that they are all `Replaceable`?

Reply:
can not confirm, I prefer leaving it for safety reason

Reply:
Please let me know if you have the confirmation that all requests enter this section is `Replacable` then I can move forward and take this check out

Reply:
I don't have such confirmation, it is my question/request for you to confirm. If there are requests that fall into that condition and are not `Replacable`, how the request is handled and how `dnfof` works for such requests?

Reply:
To confirm, I am not requesting the check to be removed, I'd like to understand how requests that are not `Replacable` (if any) are handled with `dnfof` flag `true`.

Reply:
The scope of this PR is to support `ResolveIndexAction` thus it is properly handled in the code, including `dnfof` execution path.

If there is further interest regarding how it is possible for any request that does not implement `Replacable` to enter this section, such topic can be further investigate out side the context of current PR. As for the current scope, `ResolveIndexAction` is indeed implementing `Replacable` and I can confirm as long as the request is `ResolveIndexAction`, this check will not fail